### PR TITLE
Create a model for strncpy

### LINKED ISSUE #1761: Add a model for strncpy
See: #1735

### PR DESCRIPTION
**Pull request by sschriner**
Closes #1761

### CODE REVIEW

**Comment by sschriner** [comment]
I added a directory in `tests/native/binaries` to house all binaries and c files used in c string model tests.

File: manticore/native/models.py [hunk 1]
```diff
@@ -268,3 +269,59 @@ def strcpy(state: State, dst: Union[int, BitVec], src: Union[int, BitVec]) -> Un
...
+    :param dst: destination string address
+    :param src: source string address
+    :param n: number of bytes to copy
+    :return: pointer to the dst
+    """
+
+    if issymbolic(src):
+        raise ConcretizeArgument(state.cpu, 2)
+    if issymbolic(dst):
+        raise ConcretizeArgument(state.cpu, 1)
```
Comment (ekilmer) [inline manticore/native/models.py:293]: Nit: Let's reorder this
```suggestion
    if issymbolic(dst):
        raise ConcretizeArgument(state.cpu, 1)
    if issymbolic(src):
        raise ConcretizeArgument(state.cpu, 2)
```

File: tests/native/binaries/str_model_tests/sym_strncpy_test.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#define LEN 5
+
+int main() {
+	char * A = "src or dst is A";
+	char * NA = "src or dst is not A";
```
Comment (ehennenfent) [inline tests/native/binaries/str_model_tests/sym_strncpy_test.c:9]: Did you mean to do something with `A` or `NA` here?

Comment (sschriner) [reply tests/native/binaries/str_model_tests/sym_strncpy_test.c:9]: Ah I will remove that. I had an if statement that used those char* previously. I must have forgotten to delete that with copy/paste.

**Pull request merged**